File: zircon.jvm.swing/src/test/java/org/codetome/zircon/examples/TerminalUtils.java
package org.codetome.zircon.examples;

import org.codetome.zircon.api.SwingTerminalBuilder;
import org.codetome.zircon.api.builder.TerminalBuilder;
import org.codetome.zircon.api.builder.VirtualTerminalBuilder;

public class TerminalUtils {

    public static TerminalBuilder fetchTerminalBuilder(String[] args) {
        final TerminalBuilder terminalBuilder;
        if(args.length > 0) {
            terminalBuilder = VirtualTerminalBuilder.newBuilder();
        } else {
            terminalBuilder = SwingTerminalBuilder.newBuilder();
        }
        return terminalBuilder;
    }
}

File: zircon.jvm.swing/src/test/kotlin/org/codetome/zircon/TerminalUtils.kt
package org.codetome.zircon

import org.codetome.zircon.api.SwingTerminalBuilder
import org.codetome.zircon.api.builder.grid.TerminalBuilder
import org.codetome.zircon.api.builder.grid.VirtualTerminalBuilder

object TerminalUtils {

    @JvmStatic
    fun fetchTerminalBuilder(args: Array<String>): TerminalBuilder {
        return if (args.isNotEmpty()) {
            VirtualTerminalBuilder.newBuilder()
        } else {
            SwingTerminalBuilder.newBuilder()
        }
    }
}
